sphinx/ext/intersphinx/_load.py

```diff
@@ -380,5 +380,5 @@ def _read_from_url(url: str, *, config: Config) -> IO:
     r.raw.url = r.url
     # decode content-body based on the header.
     # ref: https://github.com/psf/requests/issues/2155
-    r.raw.read = functools.partial(r.raw.read, decode_content=True)
+    r.raw.decode_content = True
```

Comment:
While most of this pull request relates only to typechecking, this line is a behaviour change, and so it's worth being careful about.

As of version v8.0.2 (most recent at time-of-writing), Sphinx [requires `requests>=2.30.0`](https://github.com/sphinx-doc/sphinx/blob/043750e406d8fe03953f5b565b36e05aef6866ec/pyproject.toml#L71).

Following that transitively, `requests` v2.30.0 (the earliest satisfying version) requires `urllib3>=1.21.1,<1.27`.

I'm solely looking at minimum versions here, but given the above, I do find that `urllib3` v1.21.1 [allows reading `decode_content` from the `HTTPResponse` object's own attributes](https://github.com/urllib3/urllib3/blob/bce03aa7f06669254e63baef54fb6278e7fa6728/urllib3/response.py#L368-L369) as a fallback when it is not provided as an argument to the `HTTPResponse.read` call.

That functionality was introduced in `urllib3` v1.10.3, and also included in v1.11 onwards.

So I believe this should be a safe adjustment to apply.